mono/metadata/image.c

```diff
@@ -2588,14 +2563,12 @@ mono_find_mempool_owner (void *ptr)
 
 		// Iterate over images within a hash
 		g_hash_table_iter_init (&iter, target);
-		while (searching && g_hash_table_iter_next(&iter, NULL, (gpointer *)&image))
+		while (!owner && g_hash_table_iter_next(&iter, NULL, (gpointer *)&image))
 		{
 			mono_image_lock (image);
```

Comment:
I just had the sad realization that this is nesting images_lock with image_lock, which is not part of the regular locking rules.

In addition to that, we're taking those two locks at the audit points, which might be violating other locking rules.
A quick check of currents uses point to this being currently ok.